Intersect.Client/Interface/Game/DescriptionWindows/DescriptionWindowBase.cs

```diff
@@ -137,32 +137,64 @@ protected void FinalizeWindow()
-            if (newX < 0)
-            {
-                newX = 0;
-            }
-            else if (newX > mContainer.Canvas.Width - mContainer.Width)
-            {
-                newX = mContainer.Canvas.Width - mContainer.Width;
-            }
+            int newX, newY;
 
-            if (newY < 0)
+            if (!Globals.Database.BindDescriptionWindowToCursor)
             {
-                newY = 0;
+                // Legacy behavior: bind description window to it's item/spell container.
+                newX = x - mContainer.Width - mContainer.Padding.Right;
+                newY = y + mContainer.Padding.Top;
+
+                // Center position when prompted by hot-bars.
+                if (mCenterOnPosition)
+                {
+                    newX += (mContainer.Width - mContainer.Padding.Right) / 2;
+                }
+
+                // Do not allow it to render outside of the screen canvas.
+                if (newX < 0)
+                {
+                    newX = 0;
+                }
+                else if (newX > mContainer.Canvas.Width - mContainer.Width)
+                {
+                    newX = mContainer.Canvas.Width - mContainer.Width;
+                }
+
+                if (newY < 0)
+                {
+                    newY = 0;
+                }
+                else if (newY > mContainer.Canvas.Height - mContainer.Height)
+                {
+                    newY = mContainer.Canvas.Height - mContainer.Height;
+                }
             }
-            else if (newY > mContainer.Canvas.Height - mContainer.Height)
+            else
             {
-                newY = mContainer.Canvas.Height - mContainer.Height;
+                // Bind description window to cursor position (ish).
+                const int posAdjust = 16;
+                newX = (int)Globals.InputManager.GetMousePosition().X + posAdjust;
+                newY = (int)Globals.InputManager.GetMousePosition().Y + posAdjust;
+
+                // Center the description window based on the cursor position when prompted by hot-bars.
+                if (mCenterOnPosition)
+                {
+                    newX = (int)Globals.InputManager.GetMousePosition().X - (mContainer.Width / 2);
+                    newY = y + mContainer.Padding.Top;
+                }
+
+                // Do not allow it to render outside of the screen canvas while based on the cursor position.
+                // Note: we no longer need to check after   (newX < 0)   nor    (newY < 0)   in this case.
+                if (newX > mContainer.Canvas.Width - mContainer.Width)
+                {
+                    newX = (int)Globals.InputManager.GetMousePosition().X - mContainer.Width - posAdjust;
+                }
+
+                if (newY > mContainer.Canvas.Height - mContainer.Height)
+                {
+                    newY = (int)Globals.InputManager.GetMousePosition().Y - mContainer.Height - posAdjust;
+                }
```

Comment:
I'd like a video/screenshots showing the behavior when it's not locked to the cursor, and ordinarily would be forced off the edge of the screen.

The current video only shows the window-locked variant when it's in the middle of the screen and doesn't show this edge case.

I am also not entirely sure that it should be bound to the cursor, it might make more sense to be bound to the thing the cursor is hovering.

I feel like having a better understanding of the full range of use cases will give us a better solution that involves less code (in particular eliminating the branch -- this code isn't an exact duplicate right now but it feels very duplicate-y, which seems to me like there may be a design flaw in particular with the edge cases).

Reply:
> The current video only shows the window-locked variant when it's in the middle of the screen and doesn't show this edge case.

Yeah, the default behaviour is kinda glitchy around the left corners sadly, i didn't meant to touch this in order to investigate a proper fix for that in a separate issue report with it's own PR later. As for the proposed windows bound to cursor enhancement, it doesn't glitches up in any corner / side of the game screen.


> I am also not entirely sure that it should be bound to the cursor, it might make more sense to be bound to the thing the cursor is hovering.

Yup, that would be way better to be fair, i just don't really know how to parse over the item/spell 's image panel positions into this, but if we were able to do so, it would be way better than binding these to the cursor.


> will give us a better solution that involves less code (in particular eliminating the branch -- this code isn't an exact duplicate right now but it feels very duplicate-y, which seems to me like there may be a design flaw in particular with the edge cases)

Yeah, it's kinda duplicate-y but we are constantly changing the same two variables in both of these cases (newX, newY), i also meant to keep it this way in order to avoid touching the previous behaviour that's been coded around here, which is, like we previously mentioned, glitchy in some scenarios.

Reply:
> I am also not entirely sure that it should be bound to the cursor, it might make more sense to be bound to the thing the cursor is hovering.

i honestly need to find out where to get the X and Y pos from them and this PR could be retaken :monocle_face: